### PR TITLE
feat(example): add a DragNDrop module in examples/

### PR DESCRIPTION
**Pull request by zarov**
This module can be added to a web page using the `<script>` tag or used
as a module in a built webapp. The user can then bind the viewer to it,
and register some file extensions to support. It is very simple, but can
be used for quick examples where people want to test files without
having to write code.

### CODE REVIEW

**Comment by zarov** [comment]
@mbredif ping if you want to test it

File: examples/js/DragNDrop.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    var _view;
+
+    function addFiles(event, files) {
+        event.preventDefault();
+
+        // Read each file
+        for (var i = 0; i < files.length; i++) {
+            var file = files[i];
+            var extension = extensionsMap[file.name.split('.').pop()];
```
Comment (gchoqueux) [inline examples/js/DragNDrop.js]: could you convert extention to low case.

File: examples/js/DragNDrop.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+                if (extension.type == _JSON) {
+                    data = JSON.parse(data);
+                } else if (extension.type == _XML) {
+                    data = new window.DOMParser().parseFromString(data, 'text/xml');
+                }
+
+                extension.parser(data, {
+                    buildExtent: true,
+                    crsIn: 'EPSG:4326',
```
Comment (gchoqueux) [inline examples/js/DragNDrop.js:71]: The projection doesn't work for all cases.

File: examples/js/DragNDrop.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                    } else {
+                        throw new Error('Mode of file not supported, please add it using DragNDrop.register');
+                    }
+
+                    _view.addLayer(layer);
+
+                    // Move the camera to the first vertex
+                    itowns.CameraUtils.animateCameraToLookAtTarget(_view, _view.camera.camera3D, {
+                        coord: new itowns.Coordinates(result.crs, result.features[0].vertices),
+                        range: 25000,
```
Comment (gchoqueux) [inline examples/js/DragNDrop.js]: could youcompute range in function the dimension of extent?

File: examples/js/DragNDrop.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        TEXT: _TEXT,
+        JSON: _JSON,
+        BINARY: _BINARY,
+        IMAGE: _IMAGE,
+        XML: _XML,
+
+        COLOR: _COLOR,
+        GEOMETRY: _GEOMETRY,
+
+        /**
```
Comment (gchoqueux) [inline examples/js/DragNDrop.js:163]: the doc isn't published?

**Review by gchoqueux — COMMENTED** [review]
The example isn't added in examples page.

File: examples/js/DragNDrop.js (hunk 2 above)
Comment (zarov) [reply examples/js/DragNDrop.js:71]: True, I don't know how to read the projection of each file. Does it needs to be added as well when registering a format ?

File: examples/js/DragNDrop.js (hunk 4 above)
Comment (zarov) [reply examples/js/DragNDrop.js:163]: Done

File: examples/js/DragNDrop.js (hunk 3 above)
Comment (zarov) [reply examples/js/DragNDrop.js]: Done

File: examples/js/DragNDrop.js (hunk 1 above)
Comment (zarov) [reply examples/js/DragNDrop.js]: Done

**Comment by zarov** [comment]
PR updated

File: examples/js/DragNDrop.js (hunk 2 above)
Comment (gchoqueux) [reply examples/js/DragNDrop.js:71]: No, but could you add this comment in the documentation and add a message in a `div` in your html example.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**